pkg/snapshot/snapshot.go

```diff
@@ -0,0 +1,369 @@
+		if policiesFile != "" {
+			policyFile, err := os.Create(policiesFile)
+			if err != nil {
+				return err
+			}
+			defer policyFile.Close()
+
+			pw := bufio.NewWriter(policyFile)
+			if err := writePolicies(policiesList, pw, e); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	// Output file will contain ApiDefinition based on specified category.
+	if category != "" {
+		category = strings.TrimSpace(category)
+		if !strings.HasPrefix(category, "#") {
+			category = fmt.Sprintf("#%s", category)
+		}
+
+		fmt.Printf("Looking for ApiDefinitions in %s category.\n", category)
+
+		for i, v := range apiDefSpecList.Apis {
+			if contains := strings.Contains(v.Name, category); !contains {
+				continue
+			}
+
+			if err := exportApiDefs(i, bw, v); err != nil {
+				return err
+			}
+		}
+
+		if err := exportPolicies(); err != nil {
+			return err
+		}
+
+		return bw.Flush()
+	}
+
+	// Output file will contain all ApiDefinitions without checking any category.
+	for i, v := range apiDefSpecList.Apis {
+		if err := exportApiDefs(i, bw, v); err != nil {
+			return err
+		}
+	}
+
+	if err := exportPolicies(); err != nil {
+		return err
+	}
+
+	return bw.Flush()
+}
+
+func createApiDef(metaName, metaNs string) tykv1alpha1.ApiDefinition {
+	meta := metav1.ObjectMeta{Name: metaName}
+	if metaNs != "" {
+		meta.Namespace = metaNs
+	}
+
+	return tykv1alpha1.ApiDefinition{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ApiDefinition",
+			APIVersion: "tyk.tyk.io/v1alpha1",
+		},
+		ObjectMeta: meta,
+		Spec:       tykv1alpha1.APIDefinitionSpec{},
+	}
+}
+
+func storeMetadata(key, name, namespace string) {
+	names[key] = name
+	nameSpaces[key] = namespace
+}
+
+func getMetadata(key string) (name, namespace string) {
+	return names[key], nameSpaces[key]
+}
+
+// fetchPolicies lists all SecurityPolicy objects form dashboard.
+func fetchPolicies(e *environmet.Env) ([]tykv1alpha1.SecurityPolicySpec, error) {
+	url := client.JoinURL(e.URL, "/api/portal/policies?p=-2")
+	method := http.MethodGet
+
+	hc := &http.Client{}
+
+	req, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Authorization", e.Auth)
+
+	res, err := hc.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	policiesResponse := &dashboard.PoliciesResponse{}
+
+	err = jsonEncoding.NewDecoder(res.Body).Decode(&policiesResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return policiesResponse.Policies, nil
+}
+
+// parseConfigData parses given ApiDefinitionSpec's ConfigData field. It checks existence of NameKey and NamespaceKey
+// keys in the ConfigData map. Returns their values if keys exist. Otherwise, returns default values for name and namespace.
+func parseConfigData(apiDefSpec *model.APIDefinitionSpec, defName string) (name, namespace string) {
+	if apiDefSpec.ConfigData == nil {
+		return defName, DefaultNs
+	}
+
+	// Parse name
+	val, ok := apiDefSpec.ConfigData.Object[NameKey]
```

Comment:
Instead of doing same thing twice for NameKey and NamespaceKey, can you store them in array and use for loop?

Reply:
Can you please elaborate on it? I couldn't fully understand. 
how can we understand that array elements belong to Name or Namespace?

Reply:
Since key name is fixed you can figure out based on that.

Or you can create a function so that we don't duplicating  the code.

Reply:
@komalsukhani added new function in bc5fd2fafe316d8209ba62045c61df99fd66e523 as you suggested. can you please check if it is okay for you?

Reply:
Looks great! Thank you.